### PR TITLE
add use_rmarkdown_template

### LINKED ISSUE #58: use_rmarkdown_template
If you want your package to provide an RMarkdown template (such as in [rtciles](https://github.com/rstudio/rticles/tree/master/inst/rmarkdown/templates), [rmdshower](https://github.com/MangoTheCat/rmdshower/tree/master/inst/rmarkdown/templates/shower) etc then there's quite a few directories you need to get right. Such as:

```
inst/rmarkdown/templates/<templatename>
inst/rmarkdown/templates/<templatename>/template.yml
inst/rmarkdown/templates/<templatename>/skeleton
inst/rmarkdown/templates/<templatename>/skeleton/skeleton.Rmd
```

A `use_rmarkdown_template` could help.

PR inbound.

### PR DESCRIPTION
**Pull request by dougmet**
Fixes #58

Feel free to ignore this or suggest as many changes as required. I have needed to make this directory structure a few times so thought usethis might be a good place to save it. 

Alternatively maybe it lives better in rmarkdown (c.f. draft)? The usethis scaffolding functions are particularly useful though.

This PR scaffolds out the directory structure and templates the `template.yml` file.

**Usage:**
```r
use_rmarkdown_template("mytemplatename")
```

it creates
```
inst/rmarkdown/templates/mytemplatename
inst/rmarkdown/templates/mytemplatename/template.yml
inst/rmarkdown/templates/mytemplatename/skeleton
inst/rmarkdown/templates/mytemplatename/skeleton/skeleton.Rmd
```

and fills in the information in `template.yml`.

**Not** accounted for:

- resources
- maybe a better default `skeleton.Rmd`?

### CODE REVIEW

File: R/rmarkdown.R [hunk 1]
```diff
@@ -0,0 +1,41 @@
+
+#' Add an RMarkdown Template
+#'
+#' @param template_dir Name of the directory the template will live in within
```
Comment (hadley) [inline R/rmarkdown.R]: Could this default to `slug(template_name)`? (and hence be the second argument?)

File: R/rmarkdown.R [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#' \code{inst/rmarkdown/templates}.
```
Comment (hadley) [inline R/rmarkdown.R]: Can you please indent second an subsequent lines with two spaces?

File: R/rmarkdown.R [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                                   template_create_dir = FALSE,
+                                   base_path = ".") {
+
+  # Process some of the inputs
+  template_create_dir <- if(isTRUE(template_create_dir)) "true" else "false"
+  template_dir <- file.path("inst", "rmarkdown", "templates", template_dir)
+
+  # Scaffold files
+  use_directory(file.path(template_dir, "skeleton"), base_path = base_path)
+  use_template("rmarkdown-template.yml", data = list(template_dir = template_dir,
```
Comment (hadley) [inline R/rmarkdown.R]: Can you please use tidyverse style for long lines? http://style.tidyverse.org/syntax.html#long-lines

File: inst/templates/rmarkdown-template.Rmd [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+output: output_format
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## R Markdown
+
+This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.
```
Comment (hadley) [inline inst/templates/rmarkdown-template.Rmd]: I think this should say something about what you should do with this template.

File: R/rmarkdown.R [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+#' use_rmarkdown_template()
+#' }
+use_rmarkdown_template <- function(template_dir = "newtemplate",
+                                   template_name = "Template Name",
+                                   template_description = "A description of the template",
+                                   template_create_dir = FALSE,
+                                   base_path = ".") {
+
+  # Process some of the inputs
+  template_create_dir <- if(isTRUE(template_create_dir)) "true" else "false"
```
Comment (hadley) [inline R/rmarkdown.R]: I think this could just be `as.character(template_create_dir)`

Comment (dougmet) [reply R/rmarkdown.R]: I was originally concerned it needed to be lowercase but I see the yaml standard is fine with upper case `TRUE` and `FALSE` so have made this change.

**Comment by dougmet** [comment]
Thanks for the review. Have made the changes.

File: R/rmarkdown.R [hunk 6]
```diff
@@ -0,0 +1,49 @@
+
+#' Add an RMarkdown Template
+#'
```
Comment (hadley) [inline R/rmarkdown.R]: Maybe a bullet list here briefly describing the files created?

I've also switch usethis to use roxygenmd, so can you please convert `\code{x}` to `` `x` ``

Then I think it's ready to merge

Comment (dougmet) [reply R/rmarkdown.R]: Great. I've had a go at those changes.

**Comment by hadley** [comment]
Perfect, thanks!

**Pull request merged**